awx/main/signals.py

```diff
@@ -637,6 +640,25 @@ def delete_inventory_for_org(sender, instance, **kwargs):
             logger.debug(e)
 
 
+@receiver(pre_delete, sender=WorkflowJobTemplateNode)
+def delete_approval_nodes(sender, instance, **kwargs):
+    if type(instance.unified_job_template) is WorkflowApprovalTemplate:
+        instance.unified_job_template.delete()
+
+
+# When setting UJT to anything other than "is approval node" - update this comment!
+@receiver(pre_save, sender=WorkflowJobTemplateNode)
+def placeholder_name(sender, instance, **kwargs):
+    try:
+        old = WorkflowJobTemplateNode.objects.get(id=instance.id)
```

Comment:
@beeankha @chrismeyersfsu 

Does this work since it's `pre_save`?  Will the `objects.get()` return the *prior* or the *new* value?

Reply:
Yep, this should all be 👍.